### ISSUE TITLE
Failing test `Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)`

### ISSUE THREAD

**Issue by radical** [dotnet/aspire#4919]
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=742191
Build error leg or test failing: Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)
Pull request: https://github.com/dotnet/aspire/pull/4910

```
Grpc.Core.RpcException : Status(StatusCode="Unavailable", Detail="Error starting gRPC call. HttpRequestException: An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake. (InvalidResponse) HttpIOException: An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake. (InvalidResponse) IOException: Unable to read data from the transport connection: Connection reset by peer. SocketException: Connection reset by peer", DebugException="System.Net.Http.HttpRequestException: An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake. (InvalidResponse)")
---- System.Net.Http.HttpRequestException : An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake. (InvalidResponse)
-------- System.Net.Http.HttpIOException : An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake. (InvalidResponse)
------------ System.IO.IOException : Unable to read data from the transport connection: Connection reset by peer.
---------------- System.Net.Sockets.SocketException : Connection reset by peer

   at Qdrant.Client.QdrantClient.CreateCollectionAsync(String collectionName, VectorsConfig vectorsConfig, UInt32 shardNumber, UInt32 replicationFactor, UInt32 writeConsistencyFactor, Boolean onDiskPayload, HnswConfigDiff hnswConfig, OptimizersConfigDiff optimizersConfig, WalConfigDiff walConfig, QuantizationConfig quantizationConfig, String initFromCollection, Nullable`1 shardingMethod, SparseVectorConfig sparseVectorsConfig, Nullable`1 timeout, CancellationToken cancellationToken)
   at Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.CreateTestDataAsync(QdrantClient qdrantClient, CancellationToken cancellationToken) in /_/tests/Aspire.Hosting.Qdrant.Tests/QdrantFunctionalTests.cs:line 62
   at Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(Boolean useVolume) in /_/tests/Aspire.Hosting.Qdrant.Tests/QdrantFunctionalTests.cs:line 134
   at Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(Boolean useVolume) in /_/tests/Aspire.Hosting.Qdrant.Tests/QdrantFunctionalTests.cs:line 140
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at System.Net.Http.Http2Connection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Grpc.Net.Client.Balancer.Internal.BalancerHttpHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)
----- Inner Stack Trace -----
   at System.Net.Http.Http2Connection.ThrowRequestAborted(Exception innerException)
   at System.Net.Http.Http2Connection.Http2Stream.TryEnsureHeaders()
   at System.Net.Http.Http2Connection.Http2Stream.ReadResponseHeadersAsync(CancellationToken cancellationToken)
   at System.Net.Http.Http2Connection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
----- Inner Stack Trace -----
   at Grpc.Net.Client.Balancer.Internal.StreamWrapper.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
   at System.Net.Http.Http2Connection.ReadFrameAsync(Boolean initialFrame)
   at System.Net.Http.Http2Connection.ProcessIncomingFramesAsync()
----- Inner Stack Trace -----
   at System.Net.Sockets.NetworkStream.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
   at Grpc.Net.Client.Balancer.Internal.StreamWrapper.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Grpc.Net.Client.Balancer.Internal.StreamWrapper.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
   at System.Net.Http.Http2Connection.ReadFrameAsync(Boolean initialFrame)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetResult(TResult result)
   at Grpc.Net.Client.Balancer.Internal.StreamWrapper.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
```

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "",
  "ErrorPattern": "Grpc.Core.RpcException .*Unavailable.*HttpRequestException: An HTTP/2 connection could not be established",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=742191
**Error message validated:** `[Grpc.Core.RpcException .*Unavailable.*HttpRequestException: An HTTP/2 connection could not be established`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 7/16/2024 1:13:25 AM UTC

cc @eerhardt

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[742651](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742651)|dotnet/aspire|[Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742651&view=ms.vss-test-web.build-test-results-tab&runId=18756356&resultId=100008)|dotnet/aspire#4927|
|[742437](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742437)|dotnet/aspire|[Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742437&view=ms.vss-test-web.build-test-results-tab&runId=18751652&resultId=100012)|dotnet/aspire#4922|
|[742471](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742471)|dotnet/aspire|[Aspire.Hosting.Milvus.Tests.MilvusFunctionalTests.WithDataVolumeShouldPersistStateBetweenUsages](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742471&view=ms.vss-test-web.build-test-results-tab&runId=18750668&resultId=100009)|dotnet/aspire#4875|
|[742459](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742459)|dotnet/aspire|[Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742459&view=ms.vss-test-web.build-test-results-tab&runId=18749764&resultId=100011)|dotnet/aspire#4763|
|[742411](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742411)|dotnet/aspire|[Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742411&view=ms.vss-test-web.build-test-results-tab&runId=18748128&resultId=100012)||
|[742409](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742409)|dotnet/aspire|[Aspire.Hosting.Milvus.Tests.MilvusFunctionalTests.WithDataVolumeShouldPersistStateBetweenUsages](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742409&view=ms.vss-test-web.build-test-results-tab&runId=18747886&resultId=100009)|dotnet/aspire#4875|
|[742382](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742382)|dotnet/aspire|[Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742382&view=ms.vss-test-web.build-test-results-tab&runId=18746736&resultId=100012)|dotnet/aspire#4915|
|[742191](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742191)|dotnet/aspire|[Aspire.Hosting.Qdrant.Tests.QdrantFunctionalTests.WithDataShouldPersistStateBetweenUsages(useVolume: False)](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742191&view=ms.vss-test-web.build-test-results-tab&runId=18740074&resultId=100012)|dotnet/aspire#4910|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|8|8|8|

**Comment by eerhardt** [comment]
I've seen this before. I think I need to add some resilience to the test.

**Comment by eerhardt** [comment]
Fixed by #4922
